snuba/query/query.py

```diff
@@ -62,6 +114,57 @@ def __init__(self, body: MutableMapping[str, Any], data_source: RelationalSource
         self.__data_source = data_source
         self.__prewhere_conditions: Sequence[Condition] = []
 
+        # New data model
+        # TODO: Provide a better typing for this.
+        self.__selected_columns = selected_columns or []
+        self.__array_join = array_join
+        self.__condition = condition
+        self.__groupby = groupby or []
+        self.__having = having
+        self.__order_by = order_by or []
+
+    def get_all_expressions(self) -> Iterable[Expression]:
+        """
+        Returns an expression container that iterates over all the expressions
+        in the query no matter which level of nesting they are at.
+        The ExpressionContainer can be used to traverse the expressions in the
+        tree.
+        """
+        return chain(*[
+            chain(*self.__selected_columns),
+            self.__array_join if self.__array_join else [],
+            self.__condition if self.__condition else [],
+            chain(*self.__groupby),
+            self.__having if self.__having else [],
+            chain(*map(lambda orderby: orderby.node, self.__order_by)),
+        ])
+
+    def transform_expression(
```

Comment:
This should probably be pluralized (`transform_expressions`) but I think `transform` might be better for consistency, unless you think that is going to be confusing.